### PR TITLE
UI Mockup

### PR DESCRIPTION
**Pull request by NickTikhonov**
This PR will add a Mockup of the user interface proposed in #11 to the app. The purpose of this is to evaluate the interface before implementing the backend component that will power the application. Proposed additions:

- [x] Implementation of every view proposed in #11 with React
- [x] Simple logic for navigating between views consistent with the design
- [x] React UI tests
- [ ] A test release of the app using GitHub releases

The final user interface will consist of components from [Primer](http://primercss.io/). This mockup will be done using [Twitter Bootstrap](http://getbootstrap.com/) to save time that would be spent writing CSS. 

CC @johndbritton @tarebyte

### CODE REVIEW

**Comment by tarebyte** [comment]
I pulled it down, and it works for me! Though we may want to add `./script/bootstrap` to `./script/run`. 

I say this because I didn't and couldn't figure out why bootstrap wasn't loading for like 15 seconds.

**Comment by NickTikhonov** [comment]
@tarebyte - bootstrap the script, or bootstrap the UI library?

**Comment by tarebyte** [comment]
> @tarebyte - bootstrap the script, or bootstrap the UI library?

The UI Library

**Comment by NickTikhonov** [comment]
@tarebyte - good point 👍  I'll do that

**Comment by NickTikhonov** [comment]
I've made an initial mockup of the archiving process 🎉  - right now the app lets you navigate between static views. I would love to get some feedback on the layout, navigation, the labels on the various elements and general usability @johndbritton @tarebyte 

screen shot 2016-07-07 at 10 41 40 am
screen shot 2016-07-07 at 10 41 53 am
screen shot 2016-07-07 at 10 42 04 am
screen shot 2016-07-07 at 10 42 20 am

In the meantime I will implement the settings view and figure out how to test React components

**Comment by johndbritton** [comment]
> GitHub Classroom Desktop

I think we should call the source repo `classroom-desktop` but the app itself should just be referred to as GitHub Classroom.

The views look like a great start.

**Comment by NickTikhonov** [comment]
I've implemented a mockup of the app and wrote tests for the front-end components. I would love some feedback on the implementation, testing, style and code structure. @btmills @atareshawty would you mind taking a 👀  at this? I have some issues I wanted to point that I wasn't able to solve:

* Every time the app gets re-rendered, I get the following error: `/Users/nicktikhonov/Drive/projects/classroom-desktop/node_modules/warning/warning.js:51 Warning: [react-router] You cannot change <Router routes>; it will be ignored`. I think this is caused when the `<Route>` components inside `<Router>` get changed, but I don't know why/where this is happening in my app. Do you know what the issue might be?

* I can't seem to figure out how to create component mocks in Jest when unit testing components that render other components. For example, the tests for `<ItemArchivePanelList>` don't work unless I unmock `<ItemArchivePanel>` and `<ItemPanel>`. Do you know what the best way of accomplishing this is?

@johndbritton @tarebyte - I would love some feedback on the design/layout, perhaps there are things I can change/improve before merging this in? I will set up auto-updates and release a version after this CR.

Thanks all in Advance :)

File: app/routes/archive/components/ItemArchivePanel.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      <p> Archiving.. </p>
+    )
+  }
+
+  const progressBarClasses = classNames("progress-bar", barColorClass)
+
+  return (
+    <ItemPanel
+      imagePath={avatarUrl}
+      title={username}
```
Comment (cyhsutw) [inline app/routes/archive/components/ItemArchivePanel.jsx]: My two cents: what do you think about prepending an `@` sign before the `username`?

Comment (cyhsutw) [inline app/routes/archive/components/ItemArchivePanel.jsx]: Ah, never mind. 
I thought there was an `@` before each username in the Classroom web app, but there isn't.

**Comment by NickTikhonov** [comment]
Alright - I think I solved the problem of testing Redux containers. I searched online to find a decent method, and the best I found involved avoiding container testing entirely by extracting all logic out into selectors. Selectors are pure functions from the state object to the values passed on to dumb components. They are more efficient - output is memoised, so a selector will only have to run once for given arguments. I'm using tools from `reselect` in order to construct reusable, efficient selectors. 

Selectors for submissions:
```JavaScript
export const all = state => state.submissions

export const num = createSelector(
  all,
  submissions => submissions.length
)

export const selected = createSelector(
  all,
  submissions => submissions.filter(each => each.selected)
)

export const numSelected = createSelector(
  selected,
  submissions => submissions.length
)
```

Container utilizing selectors:
```JavaScript
import { num, numSelected } from "../../../modules/submissions/selectors"

const mapStateToProps = (state) => ({
  title: `${numSelected(state)}/${num(state)}`,
  iconPath: "http://placehold.it/48x48",
  subtitle: "Submissions Selected"
})
```

I think there if very little need to unit test the above container given `numSelected` and `num` are tested, as we will essentially be testing that string formatting is behaving correctly.

**Comment by tarebyte** [comment]
@NickTikhonov this is looking good!

I'm not wondering if there is a better way to utilize the whitespace here?
screen shot 2016-07-21 at 21 30 31


Also this line here:
screen shot 2016-07-21 at 21 29 03

Since we have a back button, do we want to let them edit here as well?

**Comment by NickTikhonov** [comment]
🚢 this with approval from @tarebyte. The `react-router` warnings don't negatively affect the app, but I will start a separate issue to find a solution.

**Pull request merged**